telos-backend/src/models/habit.js

```diff
@@ -0,0 +1,34 @@
+const mongoose = require('mongoose');
+
+const habitSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
+  },
+  startDate: {
+    type: Date,
+    required: true,
+  },
+  // If habit is ongoing, will set default end date as the maximum possible value
+  endDate: {
+    type: Date,
+    default: new Date(8640000000000000),
+  },
+  // days of the week that this habit is set for
+  daysOfWeek: [
+    {
+      type: String,
+      required: true,
+      enum: ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'],
+    },
+  ],
+  // dates that this habit was completed
+  completedDates: [
+    {
+      type: Date,
```

Comment:
Yea I was quite confused about how to define arrays cz their documentation uses [] in the type field, but then elsewhere, all the examples have the [] outside. In the end, I decided to keep [] outside since only that syntax allows for proper use of the enum validation according to this SO: [https://stackoverflow.com/questions/40901976/mongoose-schema-enum-validation-on-array-of-strings](url)

And for the point about default, I saw that but decided to be explicit about it. 

Do you want me to change any of these two things?

Reply:
Sweet good point about enum validation. I don't mind being more explicit, so no worries there - nothing to change if the other reviewers agree